### ISSUE TITLE
Fix sentence-wise normalization in wav2vec2.0 HuggingFace recipe

### ISSUE THREAD

**Issue by gcambara** [speechbrain/speechbrain#1247]
Current advice for pretraining a wav2vec2.0 is to do zero mean and unit variance normalization for every sentence within a batch, according to the [README](https://github.com/speechbrain/speechbrain/blob/develop/recipes/CommonVoice/self-supervised-learning/wav2vec2/README.md). However, the current [implementation](https://github.com/speechbrain/speechbrain/blob/7d93a4124113a9d7b17787c3747883449f00eaff/speechbrain/lobes/models/huggingface_wav2vec.py) normalizes the whole batch, instead of each sentence separately:
```python
wav = F.layer_norm(wav, wav.shape)
```
This happens because ```wav.shape``` is used in the ```expected_shape``` argument, thus taking the batch size dimension into account. Computing the mean and std for each sentence shows that these are not 0 and 1 respectively, but they are for the whole batch:

<img width="550" alt="Screen Shot 2022-01-12 at 13 54 07" src="https://user-images.githubusercontent.com/34725661/149145466-29c6370d-f6dc-460f-92da-ea9558939772.png">

A first proposed solution would be to exclude the batch dimension:
```python
wav = F.layer_norm(wav, wav.shape[1:])
```

This indeed yields zero mean and unit variance:
<img width="664" alt="Screen Shot 2022-01-12 at 13 35 51" src="https://user-images.githubusercontent.com/34725661/149144298-c58156a9-c32c-4e7b-84ae-63a27f98950e.png">

Still, padding in the wavs is not taken into account, so this also should be implemented with a proper normalizer, as done in other recipes.

Furthermore, output features from the wav2vec2.0 encoder are normalized in the same way, batch-wise (see the [implementation](https://github.com/speechbrain/speechbrain/blob/7d93a4124113a9d7b17787c3747883449f00eaff/speechbrain/lobes/models/huggingface_wav2vec.py)):
```python
out = F.layer_norm(out, out.shape)
```

This should be checked as well, whether this is needed or not. The last transformer layer already layer normalizes the features.

On and on, if normalization is changed at the input and/or at the input, we should retrain the model and check the impact on the performance.

**Comment by TParcollet** [comment]
Thanks for the issue, indeed the layernorms should be removed from the lobes. Normalization should be applied within the recipe, but we would have to modify all the recipes and retrain the models :/

**Comment by TParcollet** [comment]
This should be fixed in SpeechBrain 0.6 #751

**Comment by Adel-Moumen** [comment]
PR #751 has been merged in `unstable-v6.0` please have a look. It will be soon in `develop`. 

I am therefore closing this issue. Feel free to reopen if needed! Thanks again. 

Best,
Adel
